collector/src/main/java/com/bitmovin/analytics/stateMachines/PlayerStateMachine.java

```diff
@@ -55,8 +52,35 @@ public void disableHeartbeat() {
         heartbeatHandler.removeCallbacksAndMessages(null);
     }
 
+    public void enableRebufferHeartbeat() {
+        heartbeatHandler.postDelayed(new Runnable() {
+            public void run() {
+                triggerHeartbeat();
+                currentRebufferingIntervalIndex = Math.min(currentRebufferingIntervalIndex + 1, rebufferingIntervals.size() - 1);
+                heartbeatHandler.postDelayed(this, rebufferingIntervals.get(currentRebufferingIntervalIndex));
+            }
+        }, rebufferingIntervals.get(currentRebufferingIntervalIndex));
+    }
+
+    public void disableRebufferHeartbeat() {
+        currentRebufferingIntervalIndex = 0;
+        heartbeatHandler.removeCallbacksAndMessages(null);
+    }
+
+    private void triggerHeartbeat() {
+        long elapsedTime = Util.getElapsedTime();
+        long elapsedTimeOnEnter = getElapsedTimeOnEnter();
+        videoTimeEnd = analytics.getPosition();
+        for (StateMachineListener listener : getListeners()) {
+            listener.onHeartbeat(elapsedTime - elapsedTimeOnEnter);
+        }
+        PlayerStateMachine.this.elapsedTimeOnEnter = elapsedTime;
```

Comment:
why did you suddenly use `PlayerStateMachine.this`?

Reply:
Removed it 👍